src/model_io/urdf/src/URDFParsingUtils.h

```diff
@@ -21,22 +21,28 @@ namespace iDynTree
 {
 
 
-bool inline stringToDouble(const std::string & inStr, double & outDouble)
+bool inline stringToDoubleWithClassicLocale(const std::string & inStr, double & outDouble)
 {
-    outDouble = std::atof(inStr.c_str());
-    return true;
+    std::istringstream ss(inStr);
+    ss.imbue(std::locale::classic());
+    ss >> outDouble;
+    return !(ss.fail());
 }
 
-bool inline stringToInt(const std::string & inStr, int & outInt)
+bool inline stringToIntWithClassicLocale(const std::string & inStr, int & outInt)
```

Comment:
Yes, for historical reason the `urdf_model/boost_replacements.h` headers is only included in Windows, and these are private functions of the library. We can think of moving this functions to a public header, but I would prefer to to that in a separate pull request, as any additional public function then needs to be maintained, documented and modified with care to avoid breaking downstream code.